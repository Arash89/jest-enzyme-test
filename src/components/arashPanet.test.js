import React from 'react';
import ArashPanel from './ArashPanel';
import {shallow} from 'enzyme';

it("Arash panel", () => {
    const wrapper = shallow(<ArashPanel />);

    expect(wrapper).toMatchSnapshot();
});