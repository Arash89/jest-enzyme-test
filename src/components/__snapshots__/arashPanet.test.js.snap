// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arash panel 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArashPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "primary",
      "children": Array [
        <PanelHeading
          bsClass="panel"
          componentClass="div"
        >
          <PanelTitle
            bsClass="panel"
            componentClass="h3"
          >
            this.props.title
          </PanelTitle>
        </PanelHeading>,
        <PanelBody
          bsClass="panel"
          collapsible={false}
        >
          <h2>
            Car: 
            Audi
            Q7
             
          </h2>
          this.props.body 
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
          >
            props
          </Button>
           
          <button
            className="salam"
          >
            run
          </button>
        </PanelBody>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "panel",
          "children": <PanelTitle
            bsClass="panel"
            componentClass="h3"
          >
            this.props.title
          </PanelTitle>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": "this.props.title",
            "componentClass": "h3",
          },
          "ref": null,
          "rendered": "this.props.title",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "panel",
          "children": Array [
            <h2>
              Car: 
              Audi
              Q7
               
            </h2>,
            "this.props.body ",
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
            >
              props
            </Button>,
            " ",
            <button
              className="salam"
            >
              run
            </button>,
          ],
          "collapsible": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Car: ",
                "Audi",
                "Q7",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Car: ",
              "Audi",
              "Q7",
              " ",
            ],
            "type": "h2",
          },
          "this.props.body ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "props",
              "disabled": false,
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "props",
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "run",
              "className": "salam",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "run",
            "type": "button",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "primary",
        "children": Array [
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              this.props.title
            </PanelTitle>
          </PanelHeading>,
          <PanelBody
            bsClass="panel"
            collapsible={false}
          >
            <h2>
              Car: 
              Audi
              Q7
               
            </h2>
            this.props.body 
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
            >
              props
            </Button>
             
            <button
              className="salam"
            >
              run
            </button>
          </PanelBody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": <PanelTitle
              bsClass="panel"
              componentClass="h3"
            >
              this.props.title
            </PanelTitle>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": "this.props.title",
              "componentClass": "h3",
            },
            "ref": null,
            "rendered": "this.props.title",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": Array [
              <h2>
                Car: 
                Audi
                Q7
                 
              </h2>,
              "this.props.body ",
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
              >
                props
              </Button>,
              " ",
              <button
                className="salam"
              >
                run
              </button>,
            ],
            "collapsible": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Car: ",
                  "Audi",
                  "Q7",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Car: ",
                "Audi",
                "Q7",
                " ",
              ],
              "type": "h2",
            },
            "this.props.body ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "props",
                "disabled": false,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "props",
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "run",
                "className": "salam",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "run",
              "type": "button",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
